### PR TITLE
docs: add MaxConnectedClusters documentation

### PR DESCRIPTION
**Pull request by thorn3r**
Adds a section to the ClusterMesh guide describing how to increase the limit of connected clusters using the `MaxConnectedClusters` option. Since there is no migration path for existing Cilium installations, a warning at the end of the section calls out that this should not be changed post-installation.

### CODE REVIEW

File: Documentation/network/clustermesh/clustermesh.rst [hunk 1]
```diff
@@ -65,6 +65,32 @@ Additional Requirements for Native-routed Datapath Modes
   firewall rules allowing pods in different clusters to reach each other on all
   ports.
 
+Cluster Identity Requirements
```
Comment (giorio94) [inline Documentation/network/clustermesh/clustermesh.rst]: I'd personally suggest to shift the focus (both section title and first paragraph) from identities to the number of clusters that can be meshed together. That's what IMHO users are interested in reading, while the cluster identity mode/requirements look more an implementation detail. For instance, I'd swap the first and second sentence of the paragraph, and reword a bit the first one to just point out that the `maxConnectedClusters` value needs to be set to the same value across the whole clustermesh (mentioning the cluster identity mode seems superfluous to me).

File: Documentation/network/clustermesh/clustermesh.rst [hunk 2]
```diff
@@ -65,6 +65,32 @@ Additional Requirements for Native-routed Datapath Modes
...
+| MaxConnectedClusters   | Maximum cluster-local identities |
++========================+============+==========+==========+
+| 255                    | 65535                            |
++------------------------+------------+----------+----------+
+| 511                    | 32767                            |
++------------------------+------------+----------+----------+
+
+.. warning::
+  ``MaxConnectedClusters`` can only be set once during Cilium installation and should not be
+  changed for existing clusters.
```
Comment (giorio94) [inline Documentation/network/clustermesh/clustermesh.rst]: I'd suggest extending a bit on the consequences of changing the `MaxConnectedClusters` value, something like connection disruption and possible incorrect enforcement of network policies.

File: Documentation/network/clustermesh/clustermesh.rst [hunk 3]
```diff
@@ -65,6 +65,32 @@ Additional Requirements for Native-routed Datapath Modes
...
+Cluster Identity Requirements
+=============================
+
+* All clusters must be configured with the same cluster identity mode using the
+  ``maxConnectedClusters`` option. By default, up to 255 clusters can be connected together. This
+  number can be increased to 511 by supplying ``maxConnectedClusters``:
+
+ * ConfigMap option ``max-connected-clusters=511``
+ * Helm option ``--set maxConnectedClusters=511``
+ * ``cilium install`` option ``--set maxConnectedClusters=511``
```
Comment (giorio94) [inline Documentation/network/clustermesh/clustermesh.rst]:
```suggestion
 * Helm option ``--set clustermesh.maxConnectedClusters=511``
 * ``cilium install`` option ``--set clustermesh.maxConnectedClusters=511``
```

**Review by giorio94 — COMMENTED** [review]
Thanks @thorn3r! I've left a couple of suggestions inline.

**Review by qmonnet — APPROVED** [review]

**Review by giorio94 — APPROVED** [review]
Looks great to me, thanks!

**Pull request merged**